implementations/micrometer-registry-stackdriver/src/main/java/io/micrometer/stackdriver/StackdriverMeterRegistry.java

```diff
@@ -551,8 +553,10 @@ Distribution distribution(HistogramSnapshot snapshot, boolean timeDomain) {
             }
 
             return Distribution.newBuilder()
+                // is the mean optional? better to not send as it is for a different time
+                // window than the histogram
                 .setMean(timeDomain ? snapshot.mean(getBaseTimeUnit()) : snapshot.mean())
-                .setCount(snapshot.count())
+                .setCount(cumulativeCount)
```

Comment:
This is the part that fixes things - `snapshot.count()` is the step-based count, which does not align with the histogram time window, while the `cumulativeCount` is taken from the histogram.